### PR TITLE
[Feature] :sparkles: Allow user to supply Moshi custom adapter

### LINKED ISSUE #609: Custom Moshi Instance
# Feature Request

## Description
My network requests utilize the following function: `.rxResponseObject(deserializable = moshiDeserializerOf(clazz = responseType))`. However, not all of my Moshi JsonAdapters are created via CodeGen. I have a custom JsonAdapter for DateTime types (specifically Instant). The response fails to be parsed with `JsonDataException: Expected BEGIN_OBJECT but was STRING at ...`. The reason being, my custom adapter is contained in a Moshi instance I have internally however, `moshiDeserializerOf` creates it's own instance of Moshi which doesn't have my custom adapter added.

## Proposed Solution
A simple change to the `moshiDeserializerOf` function as such:
```
fun <T : Any> moshiDeserializerOf(clazz: Class<T>, moshi: Moshi? = null) = object : ResponseDeserializable<T> {
    override fun deserialize(content: String): T? = (moshi ?: Moshi.Builder()
                .build())
                .adapter(clazz)
                .fromJson(content)
}
```
## Alternatives I've considered
Since `responseType` is an input provided to the interface I have with Fuel, I can conditionally route response parsing to my own Moshi instance if it is for Instant types, other proceed with the standard deserializer. This solution does not scale very well and is counter to Moshi's self management of JsonAdapters via CodeGen

### PR DESCRIPTION
**Pull request by kittinunf**
## Description

This PR allows users to supply with customization over the Moshi builder. This could be done via top-level moshi instance named `defaultMoshi`.

This supersedes #614 so it closes #614 and closes #609

## Type of change

Check all that apply

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (a change which changes the current internal or external interface)
- [ ] This change requires a documentation update

## How Has This Been Tested?

In case you did not include tests describe why you and how you have verified the
changes, with instructions so we can reproduce. If you have added comprehensive
tests for your changes, you may omit this section.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if necessary
- [ ] My changes generate no new compiler warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Inspect the bytecode viewer, including reasoning why

### CODE REVIEW

**Comment by kittinunf** [comment]
@ProjectZed Can you review this?

File: fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt [hunk 1]
```diff
@@ -18,13 +22,14 @@ inline fun <reified T : Any> Request.responseObject(handler: ResponseHandler<T>)
...
+            .adapter(clazz)
+            .fromJson(content)
 }
 
 inline fun <reified T : Any> moshiDeserializerOf(adapter: JsonAdapter<T>) = object : ResponseDeserializable<T> {
-    override fun deserialize(content: String): T? =
-            adapter.fromJson(content)
+    override fun deserialize(content: String): T? = adapter.fromJson(content)
+
+    override fun deserialize(inputStream: InputStream): T? = adapter.fromJson(Okio.buffer(Okio.source(inputStream)))
```
Comment (SleeplessByte) [inline fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt]: Do you really need `Okio` here?

Comment (iNoles) [reply fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt]: One of Moshi's JSON Adapter takes Okio.BufferedSource

Comment (SleeplessByte) [reply fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt]: Got it.

Then lgtm

**Comment by kittinunf** [comment]
I will wait for @iNoles for final approval. :)

File: fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+import com.github.kittinunf.fuel.core.ResponseHandler
 import com.github.kittinunf.fuel.core.response
 import com.github.kittinunf.result.Result
-import com.squareup.moshi.Moshi
 import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import okio.Okio
+import java.io.InputStream
+
+val defaultMoshi = Moshi.Builder()
```
Comment (kittinunf) [inline fuel-moshi/src/main/kotlin/com/github/kittinunf/fuel/moshi/FuelMoshi.kt]: I think this allow more customizability over @ProjectZed's PR

**Pull request merged**